firmware/src/consoleFunctions.cpp

```diff
@@ -0,0 +1,271 @@
+/*
+ * Brave firmware state machine for single Boron
+ * written by Heidi Fedorak, Apr 2021
+ * 
+ *  Yes, I straight up copied and pasted a function four times :-P
+ *  Reason 1: it makes the console functions easier to use, user just has to 
+ *             input an int to the correct function, and doesn't have to 
+ *             remember/type correctly a string command.
+ *  Reason 2: I don't have to spend time programming a command parser for 
+ *              things like 60 -threshold vs 100 -stillnessTime
+ *
+ *
+ *
+*/
+#include "Particle.h"
+#include "consoleFunctions.h"
+#include "flashAddresses.h"
+#include "stateMachine.h"
+#include "im21door.h"
+
+void setupConsoleFunctions(){
+
+  //particle console function declarations, belongs in setup() as per docs
+  Particle.function("Change_Initial_Timer", initial_timer_set); 
+  Particle.function("Change_Duration_Timer", duration_timer_set);     
+  Particle.function("Change_Stillness_Timer", stillness_timer_set);  
+  Particle.function("Change_INS_Threshold", ins_threshold_set);
+  Particle.function("Turn_Debugging_Publishes_On_Off", toggle_debugging_publishes);   
+  Particle.function("Change_IM21_Door_ID", im21_door_id_set); 
+
+}
+
+bool isValidIM21Id(String input) {
+
+    if(input.equals("")){
+    return false;
+    }
+
+    if(input.equals("e")){
+    return true;
+    } else {
+      if (!(input.length() == 8)){
+        return false;
+      }
+    }
+
+  return true;
+}
+
+int toggle_debugging_publishes(String command){
+
+  //default to invalid input
+  int returnFlag = -1;
+
+  //string.toInt() returns 0 if it fails, so can't distinguish between user 
+  //entering 0 vs entering bad input. So convert to char and use ascii table
+  const char* holder = command.c_str();
+
+  if(*(holder+1) != 0){
+    //any string longer than 1 char is invalid input, so
+    returnFlag = -1;
+  }
+  else if(*holder == '0'){
+    stateMachineDebugFlag = false;
+    returnFlag = 0;
+  }
+  else if(*holder == '1'){
+    stateMachineDebugFlag = true;
+    returnFlag = 1;
+  }
+  else{
+    //anything else is bad input so
+    returnFlag = -1;
+  }
+
+  return returnFlag;
+
+}
+
+//returns initial timer length if valid input is given, otherwise returns -1
+int initial_timer_set(String input){
+
+  int returnFlag = -1;
+
+  const char* holder = input.c_str();
+
+  //if e, echo the current threshold
+  if(*holder == 'e'){
+    EEPROM.get(ADDR_STATE1_MAX_TIME, state1_max_time);
+    returnFlag = state1_max_time/1000;
+  }
+  //else parse new threshold
+  else {
+    int timeout = input.toInt();
+    //increase timeout value to from seconds to ms
+    timeout = timeout*1000;
+
+    if(timeout == 0){
+        //string.toInt() returns 0 if input not an int
+        //and a threshold value of 0 makes no sense, so return -1
```

Comment:
This is unfortunate because I can see a use case for setting initial_timer to 0. In fact, I've tried it before in my testing because I don't think we actually need the inital_timer state at all with the simplified state machine. But anyways, I can see why it needs to be here, so no need to change anything.